src/serde_types.in.rs

```diff
@@ -45,4 +59,7 @@ pub struct AppConfig {
     pub token_validity: usize,
     /// Mapping of Domain -> OIDC Provider
     pub providers: HashMap<String, Provider>,
+    /// Template files
```

Comment:
I don't quite get what you're going for with this part. It makes sense to me to have a `Templates` struct, and I guess it also makes sense to have an instance of that struct in the `AppConfig`. However, does that necessarily mean we have to use `serde` for all this stuff? `impl Deserialize for Template` should just be a function `compile_template()` that's called once for each template in `default_templates()`.